### ISSUE TITLE
Android bad handling of destroy() event (orientation change, back button)

### ISSUE THREAD

**Issue by cerisara** [fyne-io/fyne#677]
On android, both a change in the screen orientation and pressing the "back" button triggers a "onDestroy" event. The expected behaviour of an android app in that case is to save its current state, and reloads it at the next onCreate event.
But this may not be properly handled so far, because it rather creates a "black screen" on the next onCreate event: it may be because all the objects have not been properly cleared at the previous "onDestroy" event (?) and/or because the state of the app has not been properly saved or restored ?

**Comment by cerisara** [comment]
It looks like both issues (back button and orientation) come from the fact that a fyne.io app "quits" too early, as soon as a onDestroy event occurs. I've checked with other vanilla gomobile app, and in their case, when the onDestroy event occurs, they do not break the main app event loop.
I don't know however when they do actually break this event loop, may be when some garbage collecting is done by android ??

Anyway, I've made a fix for these issues, as well as for the white screen, which makes the fyne.io apps work fine (back button, background, foreground, rotate...) as far as I can see on my phone.
This fix is available in my branch "blackscreen": https://github.com/cerisara/fyne/tree/blackscreen
